### PR TITLE
Gen settings attr docs

### PR DESCRIPTION
**Pull request by bernhardkaindl**
@igo95862 Referring to #31, I hope this helps with documenting the settings helper classes:

This PR is split into 3 commits:

###  tools/generate-settings-dataclasses.py: Generate attr docstrings(1/2)
   Prepare to generate docstrings for the attributes of settings_classes.
    Even though they are not parsed by python itself, they can be
    used to generate documentation for the settings dataclasses.
  
To not require manual edits after code generation, it was needed to
    implement sorting of required attributes like connection_id, -type
    and uuid before other attributes. The new functions for this are
    derived from the existing sort function for the settings_classes
    where we need the connection as first attribute of the profile
    because likewise, it is not an optional attribute.
    
The switch to enable generating the attribute docstrings is deferred
    to the next commit, to make reviewing this change above (which does
    not change the established code) and the minor change below easier:
    
There is a minor change in the established code but only regarding
    sdbus_async/networkmanager/settings/bond.py whose attributes
    are not actually required and should be optional. Making them
    required was a quick hack around some issue I don't recall the
    exact details of, and should be adressed at its source anyway.
    
This is the only code change as result of running the updated
    tool, and the next commit will enable the switch to generate
    the docstrings.
    
To aid in understanding and (in the future, reworking) the
    code for use in the new upcoming generator based on Jinja2,
    this also adds a few comments and docstrings for navigating
    the code.

###  tools/generate-settings-dataclasses.py: Generate attr docstrings(2/2)

This flips the switch to generate the docstrings

### sdbus_async/networkmanager/settings: Add generated attr docstrings

This adds the generated docstrings

### CODE REVIEW

**Comment by igo95862** [comment]
Screenshot from 2022-07-17 16-32-40

Generated documentation looks pretty good.

File: sdbus_async/networkmanager/settings/bond.py [hunk 1]
```diff
@@ -11,9 +11,11 @@
...
 
-    interface_name: str = field(
+    interface_name: Optional[str] = field(
         metadata={'dbus_name': 'interface-name', 'dbus_type': 's'},
+        default=None,
     )
-    options: Dict[str, str] = field(
+    options: Optional[Dict[str, str]] = field(
         metadata={'dbus_name': 'options', 'dbus_type': 'a{ss}'},
+        default={'Mode': 'Balance-Rr'},
```
Comment (igo95862) [inline sdbus_async/networkmanager/settings/bond.py:23]: This is causing following exception:

```
WARNING: autodoc: failed to import class 'networkmanager.NetworkManagerDeviceBluetoothInterfaceAsync' from module 'sdbus_async'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/sphinx/ext/autodoc/importer.py", line 61, in import_module
    return importlib.import_module(modname)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/igo95862/Projects/python-sdbus-root/python-sdbus-networkmanager/sdbus_async/networkmanager/__init__.py", line 159, in <module>
    from .objects import (
  File "/home/igo95862/Projects/python-sdbus-root/python-sdbus-networkmanager/sdbus_async/networkmanager/objects.py", line 70, in <module>
    from .settings.profile import ConnectionProfile
  File "/home/igo95862/Projects/python-sdbus-root/python-sdbus-networkmanager/sdbus_async/networkmanager/settings/__init__.py", line 9, in <module>
    from .bond import BondSettings
  File "/home/igo95862/Projects/python-sdbus-root/python-sdbus-networkmanager/sdbus_async/networkmanager/settings/bond.py", line 11, in <module>
    class BondSettings(NetworkManagerSettingsMixin):
  File "/usr/lib/python3.10/dataclasses.py", line 1185, in dataclass
    return wrap(cls)
  File "/usr/lib/python3.10/dataclasses.py", line 1176, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
  File "/usr/lib/python3.10/dataclasses.py", line 956, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
  File "/usr/lib/python3.10/dataclasses.py", line 813, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'dict'> for field options is not allowed: use default_factory
```

**Comment by igo95862** [comment]
Would be possible for the generator to also update `docs/options.rst` with the following template:

```
Network Manager settings
========================

.. autoclass:: sdbus_async.networkmanager.settings.EthernetSettings
    :members:

.. autoclass:: sdbus_async.networkmanager.settings.CdmaSettings
    :members:
```

**Review by igo95862 — CHANGES_REQUESTED** [review]
Thank you

File: sdbus_async/networkmanager/settings/bond.py (hunk 1 above)
Comment (bernhardkaindl) [reply sdbus_async/networkmanager/settings/bond.py]: Pushed generating `docs/options.rst` (new file, not yet added to the doctree) and a fix for this:
```py
-                    f.write(f"        default={str(default).title()},\n")
+                    if dbustype == "a{ss}" and default:
+                        default = f"field(default_factory = lambda: {default})"
+                    else:
+                        default = str(default).title()  # FALSE -> False
+                    f.write(f"        default={default},\n")
```

**Pull request merged**